File: src/main/scala/basics/coordinates.py

class Coordinate:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def distance_to(self, other_coordinate):
        # check out math.hypot for a nicer implementation!
        x_diff = self.x - other_coordinate.x
        y_diff = self.y - other_coordinate.y
        return (x_diff * x_diff + y_diff * y_diff) ** 0.5

    def as_string(self):
        return f"({self.x}, {self.y})"

def origin():
    return Coordinate(0, 0)

def find_nearest_point(reference, option1, option2):
    if reference.distance_to(option1) < reference.distance_to(option2):
        return option1
    else:
        return option2

File: src/main/scala/basics/coordinates.scala
package basics


// put your scala code here
class Coordinate(val x: Double, val y: Double) {

  /** Calculates the distance to another Coordinate
   *
   * @param other Coordinate to calculate the Distance to
   * @return distance between two coordinates
   */
  def distanceTo(other: Coordinate): Double = {
    val xDiff = x - other.x
    val yDiff = y - other.y
    Math.hypot(xDiff, yDiff)
  }

  /** Returns the Coordinate as String
   *
   * @return String of the Coordinate
   */
  def asString: String = {
    s"($x, $y)"
  }
}
  
/** Generates a Coordinate at the origin (0.0, 0.0)
 *
 * @return Coordinate with x = y = 0.0
 */
def origin: Coordinate = {
  new Coordinate(0.0, 0.0)
}

/** Returns the Coordinate that is closest to the Reference Coordinate
 *
 * @param reference Coordinate to calculate the distance to
 * @param option1   Coordinate to calculate the distance to reference and compare with option2
 * @param option2   Coordinate to calculate the distance to reference and compare with option1
 * @return Coordinate closest to reference
 */
def findNearestPoint(reference: Coordinate, option1: Coordinate, option2: Coordinate): Coordinate = {
  if (reference.distanceTo(option1) < reference.distanceTo(option2)) {
    option1
  } else {
    option2
  }
}



